Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -703,6 +712,26 @@ open class _ComposerVC<ExtraData: ExtraDataTypes>: _ViewController,
         suggestionsVC.removeFromParent()
         suggestionsVC.view.removeFromSuperview()
     }
+    
+    /// Creates and adds an attachment from the given URL to the `content`. Shows an alert if the file size exceeds the max file size.
+    /// - Parameters:
+    ///   - url: The URL of the attachment
+    ///   - type: The type of the attachment
+    open func addAttachmentToContent(from url: URL, type: AttachmentType) {
+        do {
+            let fileSize = try AttachmentFile(url: url).size
+            let maxFileSize = channelController?.client.config.maxAttachmentSize ?? 0
+            
+            if fileSize < maxFileSize {
+                let attachment = try AnyAttachmentPayload(localFileURL: url, attachmentType: type)
+                content.attachments.append(attachment)
+            } else {
+                showAttachmentExceedsMaxSizeAlert()
+            }
```

Comment:
This should throw an error, because then the function name is not right, it is doing 2 things.
- Adding the attachment to content
- Showing an alert if the max file size is reached

So there's a side effect that is not clear from the function name. I would recommend throwing an error, and the caller of this function should present the alert if it wants, not this function 🙂